outpost/pkg/ldap/utils.go

```diff
@@ -84,3 +89,31 @@ func (pi *ProviderInstance) GetUserDN(user string) string {
 func (pi *ProviderInstance) GetGroupDN(group api.Group) string {
 	return fmt.Sprintf("cn=%s,%s", group.Name, pi.GroupDN)
 }
+
+func (pi *ProviderInstance) GetUidNumber(user api.User) string {
+	return strconv.FormatInt(int64(pi.uidStartNumber + user.Pk), 10)
+}
+
+func (pi *ProviderInstance) GetGidNumber(group api.Group) string {
+	return strconv.FormatInt(int64(pi.gidStartNumber + pi.GetRIDForGroup(group.Pk)), 10)
+}
+
+func (pi *ProviderInstance) GetRIDForGroup(uid string) int32 {
+	data, err := hex.DecodeString(strings.Replace(uid, "-", "", -1))
+
+	if err != nil {
+		panic(err)
+	}
+
+	dataSize := len(data);
+
```

Comment:
No quite sure how the logic works, wouldn't it be easier to convert the UUID to its integer form and use the last few digits?

Reply:
Yep, it was a bit overcomplicated. I had grand plans for generating the gidNumber at first but scrapped 90% of it, this roundabout way is the scraps from that.

Doing it a simpler way now via hex to bigint conversion to just get the last 5 digits